src/common/com/intellij/plugins/haxe/lang/psi/HaxeGenericSpecialization.java

```diff
@@ -57,6 +59,16 @@ public boolean containsKey(@Nullable PsiElement element, String genericName) {
     return map.containsKey(getGenericKey(element, genericName));
   }
 
+  public HaxeGenericSpecialization filterInnerKeys() {
```

Comment:
The keys in a specialization look like `classname-T`.  When resolving the values of a type, we need to look up the version without the class name (e.g. 'T').  getInnerSpecialization() already existed to pull out the class name prefix.  However, the resolution code mingles new specialization lines with the inner ones, so this method is required to pull them back out.